packages/html/src/events/form.rs

```diff
@@ -23,6 +31,26 @@ pub struct FormData {
     pub files: Option<std::sync::Arc<dyn FileEngine>>,
 }
 
+fn convert_hashmap_to_json<K, V>(hashmap: &HashMap<K, V>) -> serde_json::Result<String>
+where
+    K: Serialize + std::hash::Hash + Eq,
+    V: Serialize,
+{
+    serde_json::to_string(hashmap)
+}
+
+impl FormData {
+    pub fn parse_json<T>(&self) -> Result<T, serde_json::Error>
+    where
+        T: serde::de::DeserializeOwned,
+    {
+        let parsed_json =
+            convert_hashmap_to_json(&self.values.clone()).expect("Failed to parse values to JSON");
```

Comment:
I don't think we need to clone here because you take a reference to the value directly after you clone.
```suggestion
            convert_hashmap_to_json(&self.values).expect("Failed to parse values to JSON");
```

Reply:
That is a blunder. ill remove it.